pkg/controllers/managementuser/rbac/roletemplates/crtb_handler.go

```diff
@@ -0,0 +1,145 @@
+			prtbClient:  uc.Management.Wrangler.Mgmt.ProjectRoleTemplateBinding(),
+		},
+		crbClient: uc.RBACw.ClusterRoleBinding(),
+	}
+}
+
+// OnChange ensures that the correct ClusterRoleBinding exists for the ClusterRoleTemplateBinding
+func (c *crtbHandler) OnChange(key string, crtb *v3.ClusterRoleTemplateBinding) (*v3.ClusterRoleTemplateBinding, error) {
+	if crtb == nil || crtb.DeletionTimestamp != nil {
+		return nil, nil
+	}
+
+	crb, err := buildClusterRoleBinding(crtb)
+	if err != nil {
+		return nil, err
+	}
+
+	ownerLabel := createCRTBOwnerLabel(crtb.Name)
+	currentCRBs, err := c.crbClient.List(metav1.ListOptions{LabelSelector: ownerLabel})
+	if err != nil {
+		return nil, err
+	}
+
+	// Find if there is a CRB that already exists and delete all excess CRBs
+	var matchingCRB *v1.ClusterRoleBinding
+	for _, currentCRB := range currentCRBs.Items {
+		if areClusterRoleBindingsSame(crb, &currentCRB) && matchingCRB == nil {
+			matchingCRB = &currentCRB
+			continue
+		}
+		if err := c.crbClient.Delete(currentCRB.Name, &metav1.DeleteOptions{}); err != nil {
+			return nil, err
+		}
+	}
+
+	// If we didn't find an existing CRB, create it
+	if matchingCRB == nil {
+		if _, err := c.crbClient.Create(crb); err != nil {
+			return nil, err
+		}
+	}
+
+	// Ensure a service account impersonator exists on the cluster
+	if crtb.UserName != "" {
+		if err := c.impersonationHandler.ensureServiceAccountImpersonator(crtb.UserName); err != nil {
+			return nil, fmt.Errorf("error deleting service account impersonator: %w", err)
+		}
+	}
+
+	return crtb, nil
+}
+
+// OnRemove deletes all ClusterRoleBindings owned by the ClusterRoleTemplateBinding
+func (c *crtbHandler) OnRemove(key string, crtb *v3.ClusterRoleTemplateBinding) (*v3.ClusterRoleTemplateBinding, error) {
+	lo := metav1.ListOptions{
+		LabelSelector: createCRTBOwnerLabel(crtb.Name),
+	}
+
+	crbs, err := c.crbClient.List(lo)
+	if err != nil {
+		return nil, err
+	}
+
+	var returnError error
+	for _, crb := range crbs.Items {
+		err = c.crbClient.Delete(crb.Name, &metav1.DeleteOptions{})
+		if err != nil && !apierrors.IsNotFound(err) {
+			returnError = errors.Join(returnError, err)
+		}
+	}
+
+	if crtb.UserName != "" {
+		if err = c.impersonationHandler.deleteServiceAccountImpersonator(crtb.UserName); err != nil {
+			return nil, err
+		}
+	}
+	return nil, returnError
+}
+
+// buildClusterRoleBinding returns the ClusterRoleBinding needed for a CRTB
+func buildClusterRoleBinding(crtb *v3.ClusterRoleTemplateBinding) (*v1.ClusterRoleBinding, error) {
+	ownerLabel := createCRTBOwnerLabel(crtb.Name)
+	roleRef := v1.RoleRef{
+		Kind: "ClusterRole",
+		Name: rbac.AggregatedClusterRoleNameFor(crtb.RoleTemplateName),
+	}
+
+	subject, err := rbac.BuildSubjectFromRTB(crtb)
+	if err != nil {
+		return nil, err
+	}
+
+	return &v1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "crb-",
+			Labels:       map[string]string{ownerLabel: "true"},
+		},
+		RoleRef:  roleRef,
```

Comment:
Nit-pick: I believe that to in-line these variables would make this more clear, with the exception of those fields that could find an error during their processing.